aten/src/ATen/native/mps/operations/Activation.mm

```diff
@@ -959,16 +959,23 @@ Tensor relu_mps(const Tensor& self) {
 }
 
 void elu_variants_out_mps (
-  const Tensor& self,
+  const Tensor& self_,
   const Scalar& alpha,
   const Scalar& scale,
   const Scalar& input_scale,
-  const Tensor& result,
+  const Tensor& result_,
   string func_name) {
 
   using namespace mps;
-  TORCH_CHECK(self.is_mps());
+  TORCH_CHECK(self_.is_mps());
+
+  auto self = self_;
```

Comment:
We need to generalize this code and put it in Placeholder as it will affect other ops as well. Also the output needs to be handled the same way.

Reply:
Cuda does it per op specific, e.g https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/cuda/UpSampleBilinear2d.cu#L461. We should generalize it, but we should properly test it to make sure it doesn't regress in performance in other parts.

Reply:
Currently the main priority is to make sure things are correct functionally. Fixing this one by one per-op is more time consuming when we know these are latent bugs. 
I would argue, this won't be a Performance issue on most networks which are NCHW bound. This is only an issue for NHWC case.

Reply:
I've updated the code and removed the scatter/gather for NHWC case. Now there shouldn't be any MPSGraph kernel executed  besies elu / elu_backward if the input / output are both NHWC / NCHW